### ISSUE TITLE
large meeting support

### ISSUE THREAD

**Issue by khusmann** [ottoscholten/zoomChatBot#1]
Hey Otto, great work here and thanks for posting!

I'm still poking around to see if this or something close to it will work for my purposes... thinking about adding support for larger groups -- do you have a good way to make a bunch of dummy accts join a meeting for testing these sort of scripts?

**Comment by ottoscholten** [comment]
Hey!

That is one of the pain points of developing this aspect, seeing it is hard to test. 

The only way I was able to do it was.. 
- copy the meeting url
- open icognito window, paste link.
- when it asks you to download zoom (click block/cancel), then click the link that is "open zoom" or "open meeting" again, it will give you another pop up, press (block,cancel), then another link will show up which says "open in browser".
- Enter a nickname, and you can enter the meeting without a registered account.
- repeat this, till you have at least 8 accounts.


I think 8 accounts is enough during testing, then when you finish building it, you could take the time to really create like 30 accounts. (If you do build it, then I will also do testing) There is no way to automate this process, since there are captcha's involved.

I know that the search functionality on my version of zoom was crap, but I am hoping the new update has it working (then adding this functionality shouldn't be too hard). 

Good luck! :) Also, if you end up writing your own script, id love to see it.

**Comment by khusmann** [comment]
Ah, that makes sense.

I've been playing with the zoom web client -- I'm thinking it might be easier to do a chrome extension there instead, and then i can target the things to click on with javascript... starting to get busy again tho, i'll let you know i'll let you know if i try anything.

**Comment by ottoscholten** [comment]
Good idea! Yeah I noticed that the web client wasn't very reliable for me (fairly slow). But I'd enjoy seeing how it turns out, and I'd enjoy contributing to that if it works better. (I come from a js background)

**Comment by khusmann** [comment]
Sweet!

> I come from a js background

I have less experience there, so maybe you can answer a question here -- the main challenge I see in hooking into the web client is being able to query which elements are the ones I want to click on. The challenge here is that the zoom client uses React to handle all its actions, so it looks like elements are only able to be found by their <span> or <div> class fields (rather than unique ids), which seems unreliable because multiple elements can share the same class. 

I can go in the javascript console and select elements based on their class and then initiate click events on them, so it looks like it'd prob work, but just wondering if there's a better way.

**Comment by ottoscholten** [comment]
I am not very familiar with chrome extensions and how they hook, but I think there is plenty of content out there. However, I think you are much better off just using loops (since rooms seem to be in asc order), I think hooking into React will just overcomplicate it all. :)

**Comment by saran-gangster** [comment]
i would like to contribute this project

**Comment by nelsonjchen** [comment]
I took an alternative approach that might support large meetings. 

https://github.com/nelsonjchen/HackyZoomBreakoutBot

**Comment by nelsonjchen** [comment]
I hooked into Redux for reading but I couldn't figure out how to hook into React.

**Comment by nelsonjchen** [comment]
Actually I couldn't hook anything at all come to think of it. But it's usable!

**Comment by ottoscholten** [comment]
Epic work dude! Great guide too, I will have to check this out! (I ended up building a google chrome extension, but its functionalities are different from yours). If I ever get the time, I’ll make it available too. Thanks tor sharing! 👌👌

**Comment by nelsonjchen** [comment]
@ottoscholten Oh haha, I didn't realize your bot did something totally different. I literally made my bot from what I thought your bot did by the cover.

**Comment by khusmann** [comment]
Awesome work @nelsonjchen! I'll have to check it out, it might be close to what I want to do.

@ottoscholten awesome, you went the chrome extension route! are you hooking into react element ids with it? i'd love to hear more or see the code if you're willing to share :)

**Comment by ottoscholten** [comment]
> @ottoscholten Oh haha, I didn't realize your bot did something totally different. I literally made my bot from what I thought your bot did by the cover.

Actually the functionalities are similar, just a slightly different approach. Like for me, I want people to sign up for a room and then to start all the breakout rooms at the same time. So for the format of the events I host, I don't want people hopping in and out of rooms, but rather to sign up for a session --> attend it till it is over (and repeat).

I could still recommend that you build a chrome extension (at some point). Since it lets you hook into the console, and then you can run these code snippets based on a user clicking a button, instead of users copy and pasting. Just an idea, but this is more a convenience for the user hosting, and would take more time from your side! Hopefully they'll bring an API support soon.

I'll post my code once I have time :), I do want to post it with a some sort of guide though, otherwise it can be confusing to get into! Keep up the awesome work guys, and thanks for sharing.

**Comment by nelsonjchen** [comment]
I'll probably need to figure out a way to get rid of the react devtools dependency and look at scraping the HTML for data to figure things out for a proper Chrome extension to work. 😢

**Comment by khusmann** [comment]
@nelsonjchen it looks like you can get rid of the devtools dep using something like this: https://stackoverflow.com/questions/29321742/react-getting-a-component-from-a-dom-element-for-debugging

**Comment by nelsonjchen** [comment]
Got my bot/script thingy extensionized: https://chrome.google.com/webstore/detail/breakout-room-bot-for-zoo/acfkhlojnkihdmgikmkilfjkapkemcnd?hl=en&authuser=0 Thanks for the suggestion and help!
